src/FilterFormatManager.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_migration;
+
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\filter\Entity\FilterFormat;
+use Drupal\filter\FilterPluginCollection;
+
+/**
+ * Class FilterFormatManager.
+ *
+ * Provides filter format related methods.
+ *
+ * @package Drupal\oe_migration
+ */
+class FilterFormatManager {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The filter format storage.
+   *
+   * @var \Drupal\Core\Config\Entity\ConfigEntityStorage
+   */
+  protected $filterFormatStorage;
+
+  /**
+   * The FilterFormatManager constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->setFilterFormatStorage();
+  }
+
+  /**
+   * The filterFormatStorage setter.
+   */
+  public function setFilterFormatStorage(): void {
+    try {
+      $this->filterFormatStorage = $this->entityTypeManager->getStorage('filter_format');
+    }
+    catch (PluginNotFoundException | InvalidPluginDefinitionException $e) {
+      $this->filterFormatStorage = NULL;
+    }
+  }
+
+  /**
+   * Returns a filter format by its id.
+   *
+   * @param string $id
+   *   The filter format id.
+   *
+   * @return \Drupal\filter\Entity\FilterFormat|null
+   *   The filter format object or null.
+   */
+  public function getFilterFormat(string $id): ?FilterFormat {
+    return $this->filterFormatStorage ? $this->filterFormatStorage->load($id) : NULL;
+  }
+
+  /**
+   * Returns a list of filter id's for a given filter format.
+   *
+   * @param \Drupal\filter\Entity\FilterFormat $filter_format
+   *   The filter format object.
+   *
+   * @return array
+   *   The list of filter id's for a given filter format or an empty list.
+   */
+  public function getFilterIds(FilterFormat $filter_format): array {
+    $ids = $filter_format->filters()->getInstanceIds();
+    return is_array($ids) && !empty($ids) ? array_keys($ids) : [];
+  }
+
+  /**
+   * Verifies whether a filter id is valid for a given filter format or not.
+   *
+   * @param string $filter_id
+   *   The filter id.
+   * @param \Drupal\filter\Entity\FilterFormat $filter_format
+   *   The filter format.
+   *
+   * @return bool
+   *   The answer to whether a filter id is valid for the given filter format or
+   *   not.
+   */
+  public function isValidFilterId(string $filter_id, FilterFormat $filter_format): bool {
```

Comment:
`isFilterIdValid()` is a more correct name.

Reply:
**Status:** Fixed
The method has been renamed to `isFilterIdValid()`.